module_utils/network_lsr/argument_validator.py

```diff
@@ -1806,3 +1810,57 @@ def validate_connection_one(self, mode, connections, idx):
                     idx,
                     "ip.ipv6_disabled is not supported by initscripts.",
                 )
+        # Setting ip.dns is not allowed when corresponding IP method for that
+        # nameserver is disabled
+        for nameserver in connection["ip"]["dns"]:
+            has_addrs4 = any(
+                address["family"] == socket.AF_INET
+                for address in connection["ip"]["address"]
+            )
+            if nameserver["family"] == socket.AF_INET and (
+                not connection["ip"]["dhcp4"] and not has_addrs4
+            ):
```

Comment:
Would be great to introduce a helper function that reduces the repeated code:

```suggestion
            if nameserver["family"] == socket.AF_INET and not _ipv4_enabled(connection):
```

Reply:
Helper funciton `def _ipv4_enabled(connection)` created